File: tests/leetcode/x/swift/0009.swift
import Foundation

func isPalindrome(_ x: Int) -> Bool {
    if x < 0 { return false }
    let original = x
    var n = x
    var rev = 0
    while n > 0 {
        rev = rev * 10 + (n % 10)
        n /= 10
    }
    return rev == original
}

let data = String(data: FileHandle.standardInput.readDataToEndOfFile(), encoding: .utf8) ?? ""
let tokens = data.split { $0 == " " || $0 == "\n" || $0 == "\t" || $0 == "\r" }
if !tokens.isEmpty {
    let t = Int(tokens[0])!
    let out = (0..<t).map { i in isPalindrome(Int(tokens[i + 1])!) ? "true" : "false" }
    print(out.joined(separator: "\n"))
}

File: tests/leetcode/x/scala/0009.scala
import scala.io.Source

object Main extends App {
  val tokens = Source.stdin.mkString.split("\\s+").filter(_.nonEmpty)
  if (tokens.nonEmpty) {
    val t = tokens(0).toInt
    def isPalindrome(x: Int): Boolean = {
      if (x < 0) false
      else {
        val original = x
        var n = x
        var rev = 0L
        while (n > 0) {
          rev = rev * 10 + (n % 10)
          n /= 10
        }
        rev == original
      }
    }
    print(tokens.slice(1, 1 + t).map(x => if (isPalindrome(x.toInt)) "true" else "false").mkString("\n"))
  }
}
